src/util/triplea/MapXMLCreator/MapXMLCreator.java

```diff
@@ -0,0 +1,839 @@
+				break;
+			
+			case 12:
+				CanalDefinitionsPanel.layout(this, m_stepActionPanel);
+				break;
+			
+			case 13:
+				TerritoryOwnershipPanel.layout(this, m_stepActionPanel);
+				break;
+			
+			case 14:
+				UnitPlacementsPanel.layout(this, m_stepActionPanel);
+				break;
+			
+			case 15:
+				GameSettingsPanel.layout(this, m_stepActionPanel);
+				break;
+			
+			case 16:
+				layoutNotesPanel();
+
+		default:
+			break;
+		}
+		final JFrame me = this;
+		SwingUtilities.invokeLater(new Runnable()
+		{
+			public void run()
+			{
+				me.pack();
+				m_stepActionPanel.revalidate();
+				m_stepActionPanel.repaint();
+			}
+		});
+	}
+	
+	protected void layoutNotesPanel()
+	{
+		m_stepActionPanel.setLayout(new GridBagLayout());
+		
+		final JTextArea taNotes = new JTextArea(MapXMLHelper.s_notes);
+		taNotes.addFocusListener(new FocusListener()
+		{
+			
+			@Override
+			public void focusLost(FocusEvent arg0)
+			{
+				if (!MapXMLHelper.s_notes.equals(taNotes.getText()))
+				{
+					MapXMLHelper.s_notes = taNotes.getText().trim();
+				}
+			}
+			
+			@Override
+			public void focusGained(FocusEvent arg0)
+			{
+			}
+		});
+		final JScrollPane spTaNotes = new JScrollPane(taNotes);
+		
+		Dimension size = m_stepActionPanel.getSize();
+		size.height -= 100;
+		spTaNotes.setMinimumSize(size);
+		
+		GridBagConstraints gbc_lFirstRow = new GridBagConstraints();
+		gbc_lFirstRow.insets = new Insets(0, 0, 5, 0);
+		gbc_lFirstRow.gridy = 0;
+		gbc_lFirstRow.gridx = 0;
+		gbc_lFirstRow.anchor = GridBagConstraints.NORTH;
+		gbc_lFirstRow.weightx = 1.0;
+		gbc_lFirstRow.weighty = 1.0;
+		gbc_lFirstRow.gridwidth = 3;
+		m_stepActionPanel.add(spTaNotes, gbc_lFirstRow);
+
+		GridBagConstraints gbc_lSecondRow = (GridBagConstraints) gbc_lFirstRow.clone();
+		gbc_lSecondRow.weighty = 0.0;
+		gbc_lSecondRow.insets = new Insets(0, 0, 0, 0);
+		gbc_lSecondRow.gridwidth = 1;
+		gbc_lSecondRow.gridx = 1;
+		gbc_lSecondRow.gridy = 1;
+		
+		final JButton bPreviewHTML = new JButton("Preview HTML");
+		bPreviewHTML.setPreferredSize(new Dimension(300, 30));
+		bPreviewHTML.setAction(new AbstractAction("Preview HTML")
+		{
+			private static final long serialVersionUID = -5608941822299486808L;
+			
+			public void actionPerformed(final ActionEvent event)
+			{
+				showHTML(MapXMLHelper.s_notes, "HTML Preview");
+			}
+		});
+		m_stepActionPanel.add(bPreviewHTML, gbc_lSecondRow);
+
+		GridBagConstraints gbc_lThirdRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lThirdRow.gridy = 2;
+		m_stepActionPanel.add(new JLabel("<html><big>Congratulation!</big></html>"), gbc_lThirdRow);
+		
+		GridBagConstraints gbc_lFourthRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lFourthRow.gridy = 3;
+		m_stepActionPanel.add(new JLabel("<html><nobr>You have completed all the steps for creating the map XML.</nobr></html>"), gbc_lFourthRow);
+		
+		final JButton bSave = new JButton("Save Entries to XML");

```

Comment:
Please spell out "button" instead of "b", or preferably call this "saveButton" instead. It takes a few moments to figure out what the "b" means, whether it means, "boolean", "bob", or something else until a person sees it means "button". since this is a JButton, it could have been called "jbSave", or "jSave" as well. My real point being is to avoid encoding extra data in variable names (such as data type)